SDK/MobileCenterPush/Microsoft.Azure.Mobile.Push.UWP/Push.cs

```diff
@@ -1,34 +1,63 @@
 ﻿using System;
+using System.Threading.Tasks;
 using Microsoft.Azure.Mobile.Push.Shared.Ingestion.Models;
+using Microsoft.Azure.Mobile.Utils.Synchronization;
 using Windows.Networking.PushNotifications;
 
 namespace Microsoft.Azure.Mobile.Push
 {
     public partial class Push : MobileCenterService
     {
-        private async void InstanceRegister()
+        private void InstanceRegister()
         {
-            if (Enabled)
+            if (!Enabled)
             {
-                string pushToken = (await PushNotificationChannelManager.CreatePushNotificationChannelForApplicationAsync()).Uri;
+                MobileCenterLog.Warn(MobileCenterLog.LogTag, "Push service is not enabled.");
+            }
 
-                if (!string.IsNullOrEmpty(pushToken))
-                {
-                    MobileCenterLog.Debug(MobileCenterLog.LogTag, $"Push token '{pushToken}'");
+            _stateKeeper.InvalidateState();
+            var stateSnapshot = _stateKeeper.GetStateSnapshot();
+            _mutex.Unlock();
 
-                    PushInstallationLog pushInstallationLog = new PushInstallationLog(0, null, pushToken, Guid.NewGuid());
+            var pushNotificationChannel = Task.Run(() => CreatePushNotificationChannel()).Result;
 
-                    Channel.Enqueue(pushInstallationLog);
-                }
-                else
-                {
-                    MobileCenterLog.Error(LogTag, "PushToken cannot be null or empty.");
-                }
+            try
+            {
+                _mutex.Lock(stateSnapshot);
+            }
+            catch (StatefulMutexException e)
+            {
+                MobileCenterLog.Warn(MobileCenterLog.LogTag, "Push service registering with Mobile Center backend has failed", e);
+                return;
+            }
+            finally
+            {
+                _mutex.Unlock();
+            }
+
+            var pushToken = pushNotificationChannel.Uri;
+
+            if (!string.IsNullOrEmpty(pushToken))
+            {
+                MobileCenterLog.Debug(MobileCenterLog.LogTag, $"Push token '{pushToken}'");
+
+                var pushInstallationLog = new PushInstallationLog(0, null, pushToken, Guid.NewGuid());
+
+                Channel.Enqueue(pushInstallationLog);
             }
             else
             {
-                MobileCenterLog.Warn(MobileCenterLog.LogTag, "Push service is not enable.");
+                MobileCenterLog.Error(LogTag, "Push service registering with Mobile Center backend has failed. PushToken cannot be null or empty.");
```

Comment:
PushToken is not something a user can understand, since we don't provide any details, you can just keep the first sentence here.